<launch>

    <arg name="rate" default="100.0"/>
    <arg name="prefix" default=""/>
    
    <param name="robot_description" 
        textfile="$(env ROBOTOLOGY_ROOT)/configs/CentauroConfig/urdf/centauro.urdf"/>
    
    <param name="robot_description_semantic"
        textfile="$(env ROBOTOLOGY_ROOT)/configs/CentauroConfig/srdf/centauro.srdf"/>
    
    <param name="cartesian/problem_description" 
        textfile="$(find contacts_opt)/problem/contacts_opt_stack.yaml"/>
    
    <param name="cartesian/joint_blacklist" type="yaml" 
        value="[neck_velodyne]"/>
        
    <param name="cartesian/velocity_whitelist" type="yaml" 
        value="[j_wheel_1, j_wheel_2, j_wheel_3, j_wheel_4]"/>
        
    <include file="$(find cartesian_interface)/launch/cartesio.launch">
        <arg name="rate" value="$(arg rate)"/>
        <arg name="prefix" value="$(arg prefix)"/>
    </include>    
    
    <rosparam command="load" param="cartesian/torque_offset" file="$(find contacts_opt)/config/tau_offset.yaml"/>
    
                
    <node pkg="contacts_opt"
          type="forza_giusta_node"
          name="forza_giusta_node_torque_ffwd" 
          output="screen">
          
          <arg name="rate" value="100.0"/>
          
          <remap from="~torque_offset" to="cartesian/torque_offset"/>
          
          <param name="mu" value="0.3"/>
          
          <param name="log" type="bool" value="true"/>
                   
          <param name="is_model_floating_base" value="true"/>
          <param name="model_type" value="RBDL"/>
          
          <param name="links" type="yaml" 
                 value="[wheel_1, wheel_2, wheel_3, wheel_4]"/>          
                 
          <param name="blacklist" type="yaml" 
                 value="[j_wheel_1, j_wheel_2, j_wheel_3, j_wheel_4]"/>
          
    </node>
    

     <node pkg="cartesian_interface" 
          type="force_estimation_node" 
          name="force_estimator"
          output="screen">
          
          
	<remap from="~torque_offset" to="cartesian/torque_offset"/>
          
        <param name="is_model_floating_base" value="true"/>
        <param name="model_type" value="RBDL"/>
        
        <param name="links" type="yaml" value="[wheel_1, wheel_2, wheel_3, wheel_4, arm1_8, arm2_8]"/>
        <param name="chains" type="yaml" value="[front_left_leg, front_right_leg, rear_right_leg, rear_left_leg, left_arm, right_arm]"/>
        
        <param name="wheel_1/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="wheel_2/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="wheel_3/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="wheel_4/dofs" type="yaml" value="[0, 1, 2]"/>
        <param name="arm1_8/dofs"  type="yaml" value="[0, 1, 2]"/>
        <param name="arm2_8/dofs"  type="yaml" value="[0, 1, 2]"/>
        
    </node>

</launch>
